thunder/dev_utils/utils.py

```diff
@@ -36,3 +41,77 @@
     PrimIDs.PRINT,
     PrimIDs.RETURN,
 )
+
+BenchmarkComparisonData = collections.namedtuple(
+    "BenchmarkComparisonData",
+    [
+        "nvfuser_walltime",
+        "torch_compile_walltime",
+        "nvfuser_kernel_time",
+        "torch_compile_kernel_time",
+        "nvfuser_profiler_data",
+    ],
+)
+
+
+def _benchmark_fusion_region_with_nvfuser_and_torch_compile(bsym: BoundSymbol) -> BenchmarkComparisonData:
+    """
+    Benchmark the performance of nvFuser and torch.compile for a given fusion region.
+
+    This function takes a BoundSymbol generated from nvFuser and performs the following:
+    1. Executes the fusion region using both nvFuser and torch.compile.
+    2. Measures wall time and kernel time for both implementations.
+    3. Collects profiling data for the nvFuser implementation.
+
+    Args:
+        bsym (BoundSymbol): A BoundSymbol generated from nvFuser.
+
+    Returns:
+        BenchmarkComparisonData: A named tuple containing:
+            - nvfuser_walltime: Wall time for nvFuser execution using `torch.utils.benchmark.Timer`.
+            - torch_compile_walltime: Wall time for torch.compile execution using `torch.utils.benchmark.Timer`.
+            - nvfuser_kernel_time: Kernel time for nvFuser execution using `triton.testing.do_bench`.
+            - torch_compile_kernel_time: Kernel time for torch.compile execution using `triton.testing.do_bench`.
+            - nvfuser_profiler_data: Profiling data for the nvFuser implementation by calling `fusion_defition.profile`.
+
+    .. note:: The function assumes that the fusion has been previously executed and inputs are recorded.
+    """
+    assert "nvFusion" in bsym.sym.name, "Expected the BoundSymbol to be generated from nvFuser"
+    import triton  # Import triton here as it may not be available in CPU only setting.
+
+    nvfuser_callable = bsym._call_ctx[bsym.sym.name]
+    inputs = nvfuser_callable.last_inputs
```

Comment:
would there happen to be a helper that checks if a BoundSymbol is a fusion region or not?

Reply:
We can query `bsym.sym.is_fusion` to determine if a BoundSymbol is a fusion BoundSymbol.

Also, to fetch fusion BoundSymbols, we have some utilities 

https://github.com/Lightning-AI/lightning-thunder/blob/52ee541073ddf6dce78b77ddb272621f712e3c6a/thunder/examine/__init__.py#L202

https://github.com/Lightning-AI/lightning-thunder/blob/52ee541073ddf6dce78b77ddb272621f712e3c6a/thunder/examine/__init__.py#L219